Automatic tag detection fails:

libtool: compile: unable to infer tagged configuration
libtool:   error: specify a tag with '--tag'

--- a/Makefile.comm.in
+++ b/Makefile.comm.in
@@ -42,13 +42,13 @@ CXXDEPGENFLAGS=-MM
 	$(CXX) $(CXXFLAGS) -c $<
 
 .cxx.lo:
-	@$(LIBTOOL) --mode=compile $(CXX) $(CXXFLAGS) -c $<
+	@$(LIBTOOL) --mode=compile --tag=CXX $(CXX) $(CXXFLAGS) -c $<
 
 .c.o:
 	$(CC) $(CFLAGS) -c $<
 
 .c.lo:
-	@$(LIBTOOL) --mode=compile $(CC) $(CFLAGS) -c $<
+	@$(LIBTOOL) --mode=compile --tag=CC $(CC) $(CFLAGS) -c $<
 
 .m4.cxx:
 	rm -f $@
--- a/Makefile.lib.in
+++ b/Makefile.lib.in
@@ -23,7 +23,7 @@ Makefile.lt:
 	echo 'LT_OBJS='`echo $(OBJS)|sed 's/\.o/.lo/g'` >Makefile.lt
 
 lib$(LIB).la: $(LT_OBJS)
-	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LINKFLAGS) -o lib$(LIB).la $(LT_OBJS) \
+	$(LIBTOOL) --mode=link --tag=CC $(CC) $(LDFLAGS) $(LINKFLAGS) -o lib$(LIB).la $(LT_OBJS) \
 	    -rpath $(libdir) -version-info $(LTVERSION) $(DEPLIBS)
 
 install:
--- a/Makefile.prog.in
+++ b/Makefile.prog.in
@@ -18,7 +18,7 @@ PROG:=$(shell echo "$(PROG)" | sed '@program_transform_name@')
 all: $(PROG)
 
 $(PROG): $(OBJS) $(COBJS) $(LT_LIBS)
-	$(LIBTOOL) --mode=link $(CXX) $(LINKFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ $(OBJS) $(COBJS) $(LT_LIBS) $(EXT_LIBS)
+	$(LIBTOOL) --mode=link --tag=CXX $(CXX) $(LINKFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $@ $(OBJS) $(COBJS) $(LT_LIBS) $(EXT_LIBS)
 
 install: $(PROG)
 	-test -d $(DESTDIR)$(bindir) || \
